### PR TITLE
OSDOCS-3642: Dynamic plug-in doc updates

### PR DESCRIPTION
**Pull request by opayne1**
[OSDOCS-3642](https://issues.redhat.com//browse/OSDOCS-3642): Dynamic plug-in doc updates 

Version(s):
Applies to `enterprise-4.11`+

Issue:
[OSDOCS-3642](https://issues.redhat.com//browse/OSDOCS-3642)

Link to docs preview:
http://file.rdu.redhat.com/opayne/OSDOCS-3642/web_console/dynamic-plug-ins.html#about-dynamic-plugins_dynamic-plugins

Additional information:

### CODE REVIEW

**Comment by opayne1** [comment]
@jhadvig PTAL 😄

File: modules/about-dynamic-plug-ins.adoc [hunk 1]
```diff
@@ -30,5 +30,5 @@ When creating your plug-in, follow these general guidelines:
 
 * Prefix your CSS class names with your plug-in name to avoid collisions. For example, `my-plugin_\_heading` and `my-plugin_\_icon`.
 * Maintain a consistent look, feel, and behavior with other console pages.
-* Use link:https://github.com/openshift/enhancements/blob/master/enhancements/console/dynamic-plugins.md#localization[react-i18next] for localization.
+* Follow localization guidelines when creating your plug-in.
```
Comment (jhadvig) [inline modules/about-dynamic-plug-ins.adoc:33]: Do we have those? I would at bare minimum I would link to the [dynamic-demo-plugin](https://github.com/openshift/console/tree/master/dynamic-demo-plugin#i18n).
Also the locales structure is demonstrated in there.

File: modules/adding-new-extension-dynamic-plug-in.adoc [hunk 2]
```diff
@@ -2,7 +2,35 @@
...
+        "group": "kubevirt.io",
+        "version": "v1alpha3",
+        "kind": "ExampleModel"
+      }
+    }
+  }
+]
+----
+<1> Add the extension type(s) you want to include with this plug-in. You can include multiple extensions separated with a comma. 
+<2> The `$codeRef` value should be formatted as either `moduleName.exportName` for a named export or `moduleName` for the default export. Only the plug-in’s exposed modules may be used in code references.
```
Comment (jhadvig) [inline modules/adding-new-extension-dynamic-plug-in.adoc:36]:
```suggestion
<2> The `$codeRef` value should be formatted as either `moduleName.exportName` for a named export or `moduleName` for the default export. Only the plug-in’s exported modules may be used in code references.
```

File: modules/dynamic-plug-ins-get-started.adoc [hunk 3]
```diff
@@ -6,22 +6,14 @@
 [id="getting-started-with-dynamic-plugins_{context}"]
 = Getting started with dynamic plug-ins
 
-To get started using the dynamic plug-in, you must set up your environment to write a new OpenShift Console dynamic plug-in.
+To get started using the dynamic plug-in, you must set up your environment to write a new OpenShift Console dynamic plug-in. Dynamic plug-ins are disabled by default. In order to enable, a cluster administrator will need to enable them in the `console-operator` configuration.
```
Comment (jhadvig) [inline modules/dynamic-plug-ins-get-started.adoc:9]: Would be good to demonstrate how to enable the plugin via the console-operator
Check https://github.com/openshift/console/tree/master/dynamic-demo-plugin#enabling-the-plugin

**Review by jhadvig — COMMENTED** [review]
@opayne1 thanks for the PR! 👍 
Adding couple of notes.

**Comment by opayne1** [comment]
Hi @yapei These are the doc updates for dynamic plug-in. Nothing too major, mainly just adding the extension types and a couple of other minor updates. I have removed the previously existing GH links, since technically in our doc guidelines we aren't supposed to have them. 

Could you take a look? Thank you!

**Comment by ahardin-rh** [comment]
:shipit:

File: modules/running-your-dynamic-plug-in.adoc [hunk 4]
```diff
@@ -15,7 +15,7 @@ You can run the plug-in using a local development environment. The OpenShift con
 
 .Procedure
 
-* Open two terminal windows in the local directory of your cloned repository.
+* Open two terminal windows in your local directory.
```
Comment (yapei) [inline modules/running-your-dynamic-plug-in.adoc:18]:
```
Open two terminal windows in your local directory
```
it looks like we missed some context here, what repo shall we use?

Comment (opayne1) [reply modules/running-your-dynamic-plug-in.adoc:18]: Would it be your local `openshift-console` directory or local `openshift-console/dynamic-plugin-sdk-webpack` directory?

Comment (jhadvig) [reply modules/running-your-dynamic-plug-in.adoc:18]: For Local development I would follow [these ](https://github.com/openshift/console/tree/master/dynamic-demo-plugin#local-development) that we are using for the dynamic-demo-plugin

File: modules/adding-new-extension-dynamic-plug-in.adoc [hunk 5]
```diff
@@ -2,7 +2,35 @@
...
+        "group": "kubevirt.io",
+        "version": "v1alpha3",
+        "kind": "ExampleModel"
+      }
+    }
+  }
+]
+----
+<1> Add the extension type(s) you want to include with this plug-in. You can include multiple extensions separated with a comma.
+<2> The `$codeRef` value should be formatted as either `moduleName.exportName` for a named export or `moduleName` for the default export. Only the plug-in’s exported modules may be used in code references.
```
Comment (ahardin-rh) [inline modules/adding-new-extension-dynamic-plug-in.adoc:36]: s/may/can

File: modules/disabling-plug-in-browser.adoc [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+:_content-type: PROCEDURE
+[id="disabling-your-plug-in-browser_{context}"]
+= Disabling your plug-in in the browser
+
+Console users can use the `disable-plugins` query parameter to disable specific or all dynamic plug-ins that would normally get loaded at run-time.
+
+.Procedure
+
+* To disable a specific plug-in(s), remove the plug-in you want to disable from the comma separated list of plug-in names.
```
Comment (ahardin-rh) [inline modules/disabling-plug-in-browser.adoc:13]: comma-separated list

File: modules/disabling-plug-in-browser.adoc [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+* To disable all plug-ins leave an an empty string in the `disable-plugins` query parameter.
```
Comment (ahardin-rh) [inline modules/disabling-plug-in-browser.adoc:15]: plug-ins, leave

File: modules/enabling-plug-in-browser.adoc [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+.Procedure
+
+. In the *Administration* -> *Cluster Settings* page of the web console, click the *Configuration* tab.
+
+. Click the `Console` `operator.openshift.io` configuration resource.
+
+. From there, click the *Console plugins* tab to view the dynamic plug-ins running.
+
+. In the `Status` column, click `Enable console plugin` in the pop-over which will launch the `Console plugin enablement` modal.
```
Comment (ahardin-rh) [inline modules/enabling-plug-in-browser.adoc:18]: pop-over menu?

Comment (ahardin-rh) [reply modules/enabling-plug-in-browser.adoc:18]: , which will

**Review by spadgett — COMMENTED** [review]
Hey @opayne1 @ahardin-rh Can we revisit some of the changes to the doc? I'm not sure they are accurate anymore.

File: modules/running-your-dynamic-plug-in.adoc (hunk 4 above)
Comment (spadgett) [reply modules/running-your-dynamic-plug-in.adoc]: > For Local development I would follow [these ](https://github.com/openshift/console/tree/master/dynamic-demo-plugin#local-development) that we are using for the dynamic-demo-plugin

This is no longer the recommended approach... We should use the `yarn run start-console` command as was previously documented.

**Pull request merged**